app/controllers/media_controller.rb

```diff
@@ -207,17 +211,49 @@ def inspect
   # return all media that match the search parameters
   def search
     authorize! :search, Medium.new
+
+    # get all media, then set them to only those that are visible to the current user
+    if search_params[:access].blank?
```

Comment:
If generic users manipulate their search form and add some non-blank entry for the access parameter, they will see unfiltered results, since the filtering is triggered by `filter_media`, see line 240. This should be prevented. I think they should always be filtered for non-gneric users.